src/groupeddataframe/splitapplycombine.jl

```diff
@@ -414,10 +416,45 @@ julia> combine(df -> sum(df.c), gd) # Slower variant
 
 See [`by`](@ref) for more examples.
 """
-function combine(f::Any, gd::GroupedDataFrame)
+combine(f::Function, gd::GroupedDataFrame; keepkeys::Bool=true) =
+    combine_helper(f, gd, keepkeys=keepkeys)
+combine(gd::GroupedDataFrame, f::Function; keepkeys::Bool=true) =
+    combine_helper(f, gd, keepkeys=keepkeys)
+
+function DataFrames.combine(gd::GroupedDataFrame, cs::Pair...; keepkeys::Bool=true)
+    if any(x -> first(x) isa Tuple, cs)
+        Base.depwarn("passing a Tuple as column selector is deprecated, use " *
+                     "an AbstractVector instead", :combine)
+        cs = collect(cs)
+        for (i, v) in enumerate(cs)
+            if first(v) isa Tuple
+                cs[i] = collect(first(v)) => last(v)
+            end
+        end
+    end
+    cs_norm = [normalize_selection(index(parent(gd)), c) for c in cs]
+    inverse_cs_norm = [to => (from =>fun) for (from, (fun, to)) in cs_norm]
+    @assert !isempty(inverse_cs_norm)
+    nt_cs = (; inverse_cs_norm...)
+    return combine_helper(nt_cs, gd, keepkeys=keepkeys)
+end
+
+function combine(gd::GroupedDataFrame; f...)
+    if length(f) == 0
+        throw(ArgumentError("combine(gd) is not allowed, use DataFrame(gd) " *
+                            "to combine a GroupedDataFrame into a DataFrame"))
+    else
+        Base.depwarn("combine(gd; f...) is deprecated, use " *
+                     "source_cols => fun => :target_col syntax instead", :combine)
```

Comment:
The message isn't very clear about what people have to do exactly. Also `target_col` isn't needed here, right?
```

Reply:
I was at cross what would be best to write here. `target_col` is needed here as `source_cols => fun => :target_col` is the same as `target_col = source_cols => fun` and the user had to specify it when using kwarg.

Do you have any recommendation what would be better (if not I will try to think up something)

Reply:
Maybe this:
```suggestion
        Base.depwarn("`combine(gd; target_col = source_cols => fun, ...)` is deprecated, use " *
                     "`combine(gd, source_cols => fun => :target_col, ...)` instead", :combine)
```
(Assuming I'm not missing some cases.)

Reply:
Looks good.